boards/arm/96b_argonkey/Kconfig.defconfig

```diff
@@ -40,4 +40,34 @@ config SPI_2
 
 endif # SPI
 
+if LSM6DSL
+
+choice LSM6DSL_BUS_TYPE
+	default LSM6DSL_SPI
+endchoice
+
+config LSM6DSL_SPI_MASTER_DEV_NAME
```

Comment:
Isn't it provided from DTS?

Reply:
Is it working fine the SPI from dts? 
I think I already asked this to you, but maybe I found some issue.
I can retry as I do not remember. Will I need to modify the dts.fixup file I guess.

Reply:
It's working for some parts such as BT controller for instance.
Yes, please let us know the status.

Reply:
Yes, it works. I have also used DT for SDHC over SPI.

Reply:
This might be because you used "st,lsm6dsl" as compatible, which will call i2c-device.yaml.
You might need to define a new compatible "st,lsm6dsl-spi" (or similar. i don't know how this kind of issue is handled in Linux) with matching "st,lsm6dsl-spi.yaml" that includes spi-device.yaml.
If this work, let's rename "st,lsm6dsl" to "st,lsm6dsl-i2c" (if acceptable from Linux point of view)

Reply:
Thanks @erwango 
It seems exactly the reason why I'm getting the error. Sorry, I didn't know exactly how this dts binding was working in Zephyr. So, the plan now is to provide the new st,lsm6dsl-spi bindings and modify the lsm6dsl to get configufartions from dts. Concerning the fix to lsm6dsl-i2c I would skip this for the moment as it can broke other parts of the code.

Reply:
@erwango @ydamigos 
I'm about to push a new PR which provides SPI configuration for LSM6DSL in dts.
In particular following info are provided through dts:

   CONFIG_LSM6DSL_SPI_SELECT_SLAVE
   CONFIG_LSM6DSL_SPI_MASTER_DEV_NAME
   CONFIG_LSM6DSL_SPI_BUS_FREQ   
   CONFIG_LSM6DSL_GPIO_DEV_NAME
   CONFIG_LSM6DSL_GPIO_PIN_NUM

Please note that the latter two are related to triggers (and not to SPI), but the other board that is using LSM6DSL is disco_l475_iot1 and it is NOT using triggers for LSM6DSL, in fact it builds fine.
Pls provide to me your comments about it.